### PR TITLE
Pokemon Sun and Moon WIP update

### PR DESCRIPTION
**Pull request by AustinXII**
Todo: Z-Moves, New moves, and add psychic terrain support.
Plus some little things (soul dew new boost)

### CODE REVIEW

File: calc_bc.html [hunk 1]
```diff
@@ -48,12 +49,12 @@
     <div>
         <span class="title-text">Tier(s)/Format: </span>
         <span class="tiers" title="Select the tier(s)/format.">
-            <input class="btn-input" type="radio" name="tier" id="VGC14" /><label class="btn btn-left gen-specific g6" for="VGC14">VGC14</label>
-            <input class="btn-input" type="radio" name="tier" id="Doubles" /><label class="btn btn-mid gen-specific g5 g6" for="Doubles">Doubles</label>
-            <input class="btn-input" type="radio" name="tier" id="LC" /><label class="btn btn-mid gen-specific g4 g5 g6" for="LC">LC</label>
-            <input class="btn-input" type="checkbox" name="tier" id="PU" /><label class="btn btn-mid gen-specific g6" for="PU">PU</label>
-            <input class="btn-input" type="checkbox" name="tier" id="NU" /><label class="btn btn-mid gen-specific g3 g4 g5 g6" for="NU">NU</label>
-            <input class="btn-input" type="checkbox" name="tier" id="RU" /><label class="btn btn-mid gen-specific g5 g6" for="RU">RU</label>
+            <input class="btn-input" type="radio" name="tier" id="VGC16" /><label class="btn btn-left gen-specific g7" for="VGC14">VGC16</label>
```
Comment (AustinXII) [inline calc_bc.html]: Whoops forgot to change one of them to vgc 16

File: js/damage.js [hunk 2]
```diff
@@ -413,7 +426,7 @@ function getDamageResult(attacker, defender, move, field) {
             (attacker.item === "Light Ball" && attacker.name === "Pikachu")) {
         atMods.push(0x2000);
         description.attackerItem = attacker.item;
-    } else if ((attacker.item === "Soul Dew" && (attacker.name === "Latios" || attacker.name === "Latias") && move.category === "Special") ||
+    } else if ((gen > 7 && attacker.item === "Soul Dew" && (attacker.name === "Latios" || attacker.name === "Latias") && move.category === "Special") ||
             (attacker.item === "Choice Band" && move.category === "Physical") ||
```
Comment (AustinXII) [inline js/damage.js]: gen < 7* ~_~

**Comment by Zarel** [comment]
Do you want this merged now?

File: js/damage.js [hunk 3]
```diff
@@ -516,7 +529,7 @@ function getDamageResult(attacker, defender, move, field) {
     // the random factor is applied between the crit mod and the stab mod, so don't apply anything below this until we're inside the loop
     var stabMod = 0x1000;
     if (move.type === attacker.type1 || move.type === attacker.type2) {
-        if (attacker.ability === "Adaptability") {
+        if (attacker.ability === "Adaptability" || attacker.ability === "Steelworker") {
```
Comment (Honko) [inline js/damage.js]: This doesn't seem like the right place for Steelworker. It's just a 1.5x boost to Steel-type moves, right?

File: js/damage.js [hunk 4]
```diff
@@ -321,14 +334,14 @@ function getDamageResult(attacker, defender, move, field) {
         description.isHelpingHand = true;
     }
     
-    if (isAerilate || isPixilate || isRefrigerate) {
+    if (isAerilate || isPixilate || isRefrigerate || isGalvanize || isLiquidVoice) {
```
Comment (Honko) [inline js/damage.js]: I believe Liquid Voice is just a type change, no boost.

File: js/damage.js (hunk 3 above)
Comment (AustinXII) [inline js/damage.js]: Hmm possibly I think my original source was wrong

**Comment by AustinXII** [comment]
@Zarel no I'll fix these things in a few hours, got some things to do today until then.

File: js/damage.js (hunk 3 above)
Comment (AustinXII) [inline js/damage.js]: You are right, will fix

**Pull request merged**